modifiers/allocation/modifier.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2023 Lawrence Livermore National Security, LLC and other
+# Benchpark Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import math
+from enum import Enum
+from ramble.modkit import *
+
+
+class AllocOpt(Enum):
+    # Experiment resource requests
+    N_RANKS = 1
+    N_NODES = 2
+    N_CORES_PER_TASK = 3
+    N_THREADS = 4 # number of OMP threads per task
+    N_RANKS_PER_NODE = 5
+
+    # Descriptions of resources available on systems
+    GPUS_PER_NODE = 100
+    CPUS_PER_NODE = 101
+
+
+# If we see this value, we assume the user wants us to substitute it.
+# Ramble expects n_ranks and n_nodes to be set.
+SENTINEL_UNDEFINED_VALUE = 7
+
+
+def defined_allocation_options(expander):
+    """For each possible allocation option, check if it was filled in by
+       Ramble: in that case it will be an integer.
+    """
+    defined = {}
+    for alloc_opt in AllocOpt:
+        var_def = expander.expand_var(f"{{{alloc_opt.name.lower()}}}")
```

Comment:
This appears to work for some variables, like `n_threads`, but not ones like `n_cores_per_task`/`n_ranks_per_node` (the modified `experiments/saxpy/openmp/ramble.yaml` in this PR):

```
  applications:
    saxpy:
      workloads:
        problem:
          env_vars:
            set:
              OMP_NUM_THREADS: '{n_threads}'
          variables:
            n_cores_per_task: '1'
          experiments:
            saxpy_{n}_{n_nodes}_{n_ranks}_{n_threads}:
              variables:
                n_ranks_per_node: ['8', '4']
                n_nodes: ['1', '2']
                n_threads: ['2', '4']
                n: ['512', '1024']
  applications:
    saxpy:
      workloads:
        problem:
          env_vars:
            set:
              OMP_NUM_THREADS: '{n_threads}'
          variables:
            n_cores_per_task: '1'
          experiments:
            saxpy_{n}_{n_nodes}_{n_ranks}_{n_threads}:
              variables:
                n_ranks_per_node: ['8', '4']
                n_nodes: ['1', '2']
                n_threads: ['2', '4']
                n: ['512', '1024']

```